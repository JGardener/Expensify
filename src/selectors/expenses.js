import moment from "moment";

// Get visible expenses
const getVisibleExpenses = (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        const createdAtMoment = moment(expense.createdAt);
        // startDate match will always return true
        // We gave the user an option to clear the dates, so we check that startDate exists. 
        // If there is no startDate we return true.
        // Otherwise, using isSameOrBefore, we can check to see if startDate is the same or before the createdAt time for the expense
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true ;
        // The same applied to endDateMatch
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());

        return startDateMatch && endDateMatch && textMatch;
    }).sort((a,b) => {
        if(sortBy === "date"){
            return a.createdAt < b.createdAt ? 1 : -1;
        }
        else if(sortBy === "amount"){
            return a.amount < b.amount ? 1 : -1;
        };
    })
}

export default getVisibleExpenses;