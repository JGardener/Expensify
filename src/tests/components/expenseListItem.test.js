import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';
import { expect } from '@jest/globals';

test("Should render expense information with a link attached to the expense description", () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/> );
    expect(wrapper).toMatchSnapshot();
})
