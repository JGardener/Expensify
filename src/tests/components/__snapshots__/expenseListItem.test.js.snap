// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense information with a link attached to the expense description 1`] = `
<div>
  <[object Object]
    to="edit/1"
  >
    <h3>
      Gum
    </h3>
  </[object Object]>
  <p>
    195
     - 
    0
  </p>
</div>
`;
